packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -248,27 +259,94 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
             visualizeElements(hilitedIds, stringToColor(row.id ?? ""));
             allIds = allIds.concat(hilitedIds);
           } else {
-            const ids: string[] = await fetchIdsFromQuery(
-              row.query ?? "",
-              iModelConnection,
-            );
-            const hiliteIds = await visualizeElementsById(
-              ids,
-              stringToColor(row.id ?? ""),
-              iModelConnection,
-            );
-            hilitedElements.current.set(row.query ?? "", hiliteIds);
+            try {
+              const ids: string[] = await fetchIdsFromQuery(
+                row.query ?? "",
+                iModelConnection,
+              );
+              const hiliteIds = await visualizeElementsById(
+                ids,
+                stringToColor(row.id ?? ""),
+                iModelConnection,
+              );
+              hilitedElements.current.set(row.query ?? "", hiliteIds);
 
-            allIds = allIds.concat(ids);
+              allIds = allIds.concat(ids);
+            } catch {
+              const index = groups.findIndex((group) => group.id === row.id);
+              setSelectedRows((rowIds) => {
+                const selectedRowIds = { ...rowIds };
+                delete selectedRowIds[index];
```

Comment:
might be me, but i dont like the use of `delete` here, couldnt you just use Array.prototype.filter?

Reply:
I looked into again. I feel like this will be a pain since the selectedRowIds aren't arrays. I just remembered that react-table uses a Record with the properties being the row id's.